### PR TITLE
Use an OSGi declarative service instead of IStartup #156

### PR DESCRIPTION
**Pull request by BeckerFrank**
Replace the IStartup and org.eclipse.ui.startup extension
point by a declarative OSGi service component listening on
UIEvents.UILifeCycle.APP_STARTUP_COMPLETE.

### CODE REVIEW

**Comment by merks** [comment]
Those look like bogus dash failures.  😢

**Comment by ruspl-afed** [comment]
There are still failures in dash

**Comment by BeckerFrank** [comment]
From the log:

[INFO] License information could not be automatically verified for the following content:
[INFO] 
[INFO] p2/orbit/p2.eclipse.plugin/org.eclipse.launchbar.core/2.5.100.202302061541
[INFO] p2/orbit/p2.eclipse.plugin/org.eclipse.launchbar.ui/2.5.100.202303140100
[INFO] p2/orbit/p2.eclipse.plugin/org.eclipse.tools.templates.core/1.3.0.202211062329
[INFO] p2/orbit/p2.eclipse.plugin/org.eclipse.tools.templates.freemarker/1.3.0.202211062329
[INFO] p2/orbit/p2.eclipse.plugin/org.eclipse.tools.templates.ui/1.4.0.202211062329
[INFO] 
[INFO] This content is either not correctly mapped by the system, or requires review.

but I only found references to org.eclipse.launchbar or org.eclipse.tools.templates.freemarker in old target platform files.

- mylyn-e4.6.target
- mylyn-e4.7.target
- mylyn-e4.8.target
- mylyn-emaintenance.target
- mylyn-estaging.target

**Comment by ruspl-afed** [comment]
Actually I have no idea now to procced further with broken Dash.
The project activity is actually blocked.
I suggest to disable Dash action until we can get reply from IP team. 
WDYT @BeckerFrank @wimjongman @merks ?

**Comment by merks** [comment]
I run builds with a parameter to disable failure conditionally:
```
-Ddash.fail=$DASH_STRICT
```
So I believe that using `-Ddash.fail=false` will still do the checking, still report on the problems, and still open new issues as needed, but will not fail the build.   I'm not sure about the opening CQ thing in this case; I run dash as a regular part of the build for my Orbit-like stuff...

I believe @waynebeaton recommends not failing builds in such a way...

The above issues actually look a bogus hit.  Issues requiring manual intervention are taking a bit longer latest to clear the backlog the last while...

**Comment by ruspl-afed** [comment]
> and still open new issues as needed

are you sure regarding this? I thought one need to do it explicitly via `/request-license-review`.

> I believe @waynebeaton recommends not failing builds in such a way...

Yes, true, he does. My concern is that if it doesn't fail, no one will take care to view the log of a check that didn't fail.

**Comment by merks** [comment]
It's a legitimate concern.  This is why I parameterize my build, but that doesn't help here.   So you don't have many alternatives:  Wait for goodness knows how long, or risk not noticing ip issues...

**Comment by ruspl-afed** [comment]
@BeckerFrank .target removals seems to be unrelated

File: mylyn.context/org.eclipse.mylyn.team.ui/plugin.xml [hunk 1]
```diff
@@ -17,14 +17,15 @@
    <extension-point id="commitTemplates" name="Commit Template Handlers" schema="schema/commitTemplates.exsd"/>
     
     <!-- needed to update change sets -->
+<!--
```
Comment (ruspl-afed) [inline mylyn.context/org.eclipse.mylyn.team.ui/plugin.xml:20]: Let's just remove this extension if we don't need it

**Review by ruspl-afed — CHANGES_REQUESTED** [review]
Please consider removing commented extensions

File: mylyn.context/org.eclipse.mylyn.team.ui/plugin.xml (hunk 1 above)
Comment (BeckerFrank) [reply mylyn.context/org.eclipse.mylyn.team.ui/plugin.xml]: fixed with new commit

File: mylyn.tasks/org.eclipse.mylyn.tasks.ui/plugin.xml [hunk 2]
```diff
@@ -56,14 +56,14 @@
         <category name="%newWizards.category.name" id="org.eclipse.mylyn.ui.wizards.new"/>
     </extension>
 
-    <extension
+<!--    <extension
```
[the remark this answers is not in the document]
Comment (BeckerFrank) [reply mylyn.tasks/org.eclipse.mylyn.tasks.ui/plugin.xml]: fixed with new commit

**Review by ruspl-afed — APPROVED** [review]

**Pull request merged**